"Table 'jhipster.jhi_user' doesn't exist" error on new app

## Overview

After creating a new app using MySQL and OAuth2 on Linux, the app is inaccessible because the user always receives an authentication error.

The logs report `Table 'jhipster.jhi_user' doesn't exist` even though the table was created correctly upon application deployment.

The problem seems to be associated with the case of the table name. The query is using lowercase while the table was created using uppercase. Using the same case in the query and the table name seems to fix the issue.

This didn't happen in previous versions of JHipster.

## JHipster version(s)

2.21.0

## JHipster configuration
```json
`{
  "generator-jhipster": {
    "baseName": "jhipster",
    "packageName": "com.piedrahitac.jhipster",
    "packageFolder": "com/piedrahitac/jhipster",
    "authenticationType": "oauth2",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": "no",
    "websocket": "no",
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": "no",
    "useSass": true,
    "buildTool": "maven",
    "frontendBuilder": "grunt",
    "javaVersion": "8",
    "enableTranslation": true,
    "rememberMeKey": "obscured"
  }
}`
```
## Browsers and OS

Chrome and Firefox on Linux (Fedora 22)

## Reproduce the error

Create a brand new JHipster app, attempt to login and check the log.

This `;DATABASE_TO_UPPER=FALSE` causes:

`[ERROR] org.hibernate.tool.hbm2ddl.SchemaValidator - HHH000319: Could not get database metadata
org.h2.jdbc.JdbcSQLException: Schema "information_schema" not found; SQL statement:
select sequence_name from information_schema.sequences [90079-187]`

when running tests.

--- a/app/templates/src/main/java/package/domain/_Authority.java
+++ b/app/templates/src/main/java/package/domain/_Authority.java
@@ -16,9 +16,9 @@ import java.io.Serializable;
  * An authority (a security role) used by Spring Security.
  */<% if (databaseType == 'sql') { %>
 @Entity
-@Table(name = "JHI_AUTHORITY")<% if (hibernateCache != 'no') { %>
+@Table(name = "jhi_authority")<% if (hibernateCache != 'no') { %>
 @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)<% } %><% } %><% if (databaseType == 'mongodb') { %>
-@Document(collection = "JHI_AUTHORITY")<% } %>
+@Document(collection = "jhi_authority")<% } %>
 public class Authority implements Serializable {
 
     @NotNull

--- a/app/templates/src/main/java/package/domain/_PersistentAuditEvent.java
+++ b/app/templates/src/main/java/package/domain/_PersistentAuditEvent.java
@@ -15,8 +15,8 @@ import java.util.Map;
  * @see org.springframework.boot.actuate.audit.AuditEvent
  */<% if (databaseType == 'sql') { %>
 @Entity
-@Table(name = "JHI_PERSISTENT_AUDIT_EVENT")<% } %><% if (databaseType == 'mongodb') { %>
-@Document(collection = "JHI_PERSISTENT_AUDIT_EVENT")<% } %>
+@Table(name = "jhi_persistent_audit_event")<% } %><% if (databaseType == 'mongodb') { %>
+@Document(collection = "jhi_persistent_audit_event")<% } %>
 public class PersistentAuditEvent  {
 
     @Id<% if (databaseType == 'sql') { %>
@@ -40,7 +40,7 @@ public class PersistentAuditEvent  {
     @ElementCollection
     @MapKeyColumn(name="name")
     @Column(name="value")
-    @CollectionTable(name="JHI_PERSISTENT_AUDIT_EVT_DATA", joinColumns=@JoinColumn(name="event_id"))<% } %>
+    @CollectionTable(name="jhi_persistent_audit_evt_data", joinColumns=@JoinColumn(name="event_id"))<% } %>
     private Map<String, String> data = new HashMap<>();
 <% if (databaseType == 'sql') { %>
     public Long getId() {

--- a/app/templates/src/main/java/package/domain/_PersistentToken.java
+++ b/app/templates/src/main/java/package/domain/_PersistentToken.java
@@ -28,9 +28,9 @@ import java.util.Date;<% } %>
  * @see <%=packageName%>.security.CustomPersistentRememberMeServices
  */<% if (databaseType == 'sql') { %>
 @Entity
-@Table(name = "JHI_PERSISTENT_TOKEN")<% } %><% if (hibernateCache != 'no' && databaseType == 'sql') { %>
+@Table(name = "jhi_persistent_token")<% } %><% if (hibernateCache != 'no' && databaseType == 'sql') { %>
 @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)<% } %><% if (databaseType == 'mongodb') { %>
-@Document(collection = "JHI_PERSISTENT_TOKEN")<% } %><% if (databaseType == 'cassandra') { %>
+@Document(collection = "jhi_persistent_token")<% } %><% if (databaseType == 'cassandra') { %>
 @Table(name = "persistent_token")<% } %>
 public class PersistentToken implements Serializable {
 

--- a/app/templates/src/main/java/package/domain/_User.java
+++ b/app/templates/src/main/java/package/domain/_User.java
@@ -32,9 +32,9 @@ import org.joda.time.DateTime;<% } %>
  * A user.
  */<% if (databaseType == 'sql') { %>
 @Entity
-@Table(name = "JHI_USER")<% } %><% if (hibernateCache != 'no' && databaseType == 'sql') { %>
+@Table(name = "jhi_user")<% } %><% if (hibernateCache != 'no' && databaseType == 'sql') { %>
 @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)<% } %><% if (databaseType == 'mongodb') { %>
-@Document(collection = "JHI_USER")<% } %><% if (databaseType == 'cassandra') { %>
+@Document(collection = "jhi_user")<% } %><% if (databaseType == 'cassandra') { %>
 @Table(name = "user")<% } %><% if (searchEngine == 'elasticsearch') { %>
 @Document(indexName="user")<% } %>
 public class User<% if (databaseType == 'sql' || databaseType == 'mongodb') { %> extends AbstractAuditingEntity<% } %> implements Serializable {
@@ -111,7 +111,7 @@ public class User<% if (databaseType == 'sql' || databaseType == 'mongodb') { %>
     @JsonIgnore<% if (databaseType == 'sql') { %>
     @ManyToMany
     @JoinTable(
-            name = "JHI_USER_AUTHORITY",
+            name = "jhi_user_authority",
             joinColumns = {@JoinColumn(name = "user_id", referencedColumnName = "id")},
             inverseJoinColumns = {@JoinColumn(name = "authority_name", referencedColumnName = "name")})<% if (hibernateCache != 'no') { %>
     @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)<% } %><% } %><% if (databaseType == 'sql' || databaseType == 'mongodb') { %>

--- a/app/templates/src/main/resources/config/liquibase/changelog/_initial_schema.xml
+++ b/app/templates/src/main/resources/config/liquibase/changelog/_initial_schema.xml
@@ -2,7 +2,8 @@
 <databaseChangeLog
     xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd"
+    objectQuotingStrategy="QUOTE_ALL_OBJECTS">
 
     <property name="now" value="now()" dbms="mysql,h2"/>
     <property name="now" value="current_timestamp" dbms="postgresql"/>
@@ -20,9 +21,9 @@
         The initial schema has the '00000000000001' id, so that it is over-written if we re-generate it.
     -->
     <changeSet id="00000000000001" author="jhipster"><% if(prodDatabaseType === 'oracle') {%>
-        <createSequence sequenceName="SEQ_USER" startValue="1000" incrementBy="1"/><% } %>
-        <createTable tableName="JHI_USER">
-            <column name="id" type="bigint" autoIncrement="${autoIncrement}" <% if(prodDatabaseType === 'oracle') {%>defaultValueComputed="SEQ_USER.NEXTVAL"<% } %>>
+        <createSequence sequenceName="seq_user" startValue="1000" incrementBy="1"/><% } %>
+        <createTable tableName="jhi_user">
+            <column name="id" type="bigint" autoIncrement="${autoIncrement}" <% if(prodDatabaseType === 'oracle') {%>defaultValueComputed="seq_user.NEXTVAL"<% } %>>
                 <constraints primaryKey="true" nullable="false"/>
             </column>
             <column name="login" type="varchar(50)">
@@ -54,24 +55,24 @@
         </createTable>
         <% if(prodDatabaseType != 'oracle') {%>
         <createIndex indexName="idx_user_login"
-            tableName="JHI_USER"
+            tableName="jhi_user"
             unique="true">
             <column name="login" type="varchar(50)"/>
         </createIndex>
 
         <createIndex indexName="idx_user_email"
-            tableName="JHI_USER"
+            tableName="jhi_user"
             unique="true">
             <column name="email" type="varchar(100)"/>
         </createIndex>
         <% } %>
-        <createTable tableName="JHI_AUTHORITY">
+        <createTable tableName="jhi_authority">
             <column name="name" type="varchar(50)">
                 <constraints primaryKey="true" nullable="false"/>
             </column>
         </createTable>
 
-        <createTable tableName="JHI_USER_AUTHORITY">
+        <createTable tableName="jhi_user_authority">
             <column name="user_id" type="bigint">
                 <constraints nullable="false"/>
             </column>
@@ -80,9 +81,9 @@
             </column>
         </createTable>
 
-        <addPrimaryKey columnNames="user_id, authority_name" tableName="JHI_USER_AUTHORITY"/><% if (authenticationType == 'session') { %>
+        <addPrimaryKey columnNames="user_id, authority_name" tableName="jhi_user_authority"/><% if (authenticationType == 'session') { %>
 
-        <createTable tableName="JHI_PERSISTENT_TOKEN">
+        <createTable tableName="jhi_persistent_token">
             <column name="series" type="varchar(255)">
                 <constraints primaryKey="true" nullable="false"/>
             </column>
@@ -96,45 +97,45 @@
         </createTable><% } %>
 
         <addForeignKeyConstraint baseColumnNames="authority_name"
-                                 baseTableName="JHI_USER_AUTHORITY"
+                                 baseTableName="jhi_user_authority"
                                  constraintName="fk_authority_name"
                                  referencedColumnNames="name"
-                                 referencedTableName="JHI_AUTHORITY"/>
+                                 referencedTableName="jhi_authority"/>
 
         <addForeignKeyConstraint baseColumnNames="user_id"
-                                 baseTableName="JHI_USER_AUTHORITY"
+                                 baseTableName="jhi_user_authority"
                                  constraintName="fk_user_id"
                                  referencedColumnNames="id"
-                                 referencedTableName="JHI_USER"/><% if (authenticationType == 'session') { %>
+                                 referencedTableName="jhi_user"/><% if (authenticationType == 'session') { %>
 
         <addForeignKeyConstraint baseColumnNames="user_id"
-                                 baseTableName="JHI_PERSISTENT_TOKEN"
+                                 baseTableName="jhi_persistent_token"
                                  constraintName="fk_user_persistent_token"
                                  referencedColumnNames="id"
-                                 referencedTableName="JHI_USER"/><% } %>
+                                 referencedTableName="jhi_user"/><% } %>
 
         <loadData encoding="UTF-8"
                   file="config/liquibase/users.csv"
                   separator=";"
-                  tableName="JHI_USER">
+                  tableName="jhi_user">
             <column name="activated" type="boolean"/>
             <column name="created_date" type="timestamp"/>
         </loadData>
-        <dropDefaultValue tableName="JHI_USER" columnName="created_date" columnDataType="datetime"/>
+        <dropDefaultValue tableName="jhi_user" columnName="created_date" columnDataType="datetime"/>
 
         <loadData encoding="UTF-8"
                   file="config/liquibase/authorities.csv"
                   separator=";"
-                  tableName="JHI_AUTHORITY"/>
+                  tableName="jhi_authority"/>
 
         <loadData encoding="UTF-8"
                   file="config/liquibase/users_authorities.csv"
                   separator=";"
-                  tableName="JHI_USER_AUTHORITY"/>
+                  tableName="jhi_user_authority"/>
                   <% if(prodDatabaseType === 'oracle') {%>
-        <createSequence sequenceName="SEQ_PERSISTENT_AUDIT_EVENT" startValue="1000" incrementBy="1"/><% } %>
-        <createTable tableName="JHI_PERSISTENT_AUDIT_EVENT">
-            <column name="event_id" type="bigint" autoIncrement="${autoIncrement}" <% if(prodDatabaseType === 'oracle') {%>defaultValueComputed="SEQ_PERSISTENT_AUDIT_EVENT.NEXTVAL"<% } %>>
+        <createSequence sequenceName="seq_persistent_audit_event" startValue="1000" incrementBy="1"/><% } %>
+        <createTable tableName="jhi_persistent_audit_event">
+            <column name="event_id" type="bigint" autoIncrement="${autoIncrement}" <% if(prodDatabaseType === 'oracle') {%>defaultValueComputed="seq_persistent_audit_event.NEXTVAL"<% } %>>
                 <constraints primaryKey="true" nullable="false"/>
             </column>
             <column name="principal" type="varchar(255)">
@@ -144,7 +145,7 @@
             <column name="event_type" type="varchar(255)"/>
         </createTable>
 
-        <createTable tableName="JHI_PERSISTENT_AUDIT_EVT_DATA">
+        <createTable tableName="jhi_persistent_audit_evt_data">
             <column name="event_id" type="bigint">
                 <constraints nullable="false"/>
             </column>
@@ -153,26 +154,26 @@
             </column>
             <column name="value" type="varchar(255)"/>
         </createTable>
-        <addPrimaryKey columnNames="event_id, name" tableName="JHI_PERSISTENT_AUDIT_EVT_DATA"/>
+        <addPrimaryKey columnNames="event_id, name" tableName="jhi_persistent_audit_evt_data"/>
 
         <createIndex indexName="idx_persistent_audit_event"
-                     tableName="JHI_PERSISTENT_AUDIT_EVENT"
+                     tableName="jhi_persistent_audit_event"
                      unique="false">
             <column name="principal" type="varchar(255)"/>
             <column name="event_date" type="timestamp"/>
         </createIndex>
 
         <createIndex indexName="idx_persistent_audit_evt_data"
-                     tableName="JHI_PERSISTENT_AUDIT_EVT_DATA"
+                     tableName="jhi_persistent_audit_evt_data"
                      unique="false">
             <column name="event_id" type="bigint"/>
         </createIndex>
 
         <addForeignKeyConstraint baseColumnNames="event_id"
-                                 baseTableName="JHI_PERSISTENT_AUDIT_EVT_DATA"
+                                 baseTableName="jhi_persistent_audit_evt_data"
                                  constraintName="fk_evt_pers_audit_evt_data"
                                  referencedColumnNames="event_id"
-                                 referencedTableName="JHI_PERSISTENT_AUDIT_EVENT"/>
+                                 referencedTableName="jhi_persistent_audit_event"/>
     </changeSet>
     <% if (authenticationType == 'oauth2') { %>
     <!-- Manage the OAuth authentication -->
@@ -205,7 +206,7 @@
             baseTableName="oauth_client_token"
             constraintName="fk_oauth_client_tokn_user_name"
             referencedColumnNames="login"
-            referencedTableName="JHI_USER"/>
+            referencedTableName="jhi_user"/>
 
         <createTable tableName="oauth_access_token">
             <column name="token_id" type="varchar(255)"/>
@@ -223,7 +224,7 @@
             baseTableName="oauth_access_token"
             constraintName="fk_oauth_access_tokn_user_name"
             referencedColumnNames="login"
-            referencedTableName="JHI_USER"/>
+            referencedTableName="jhi_user"/>
 
         <createTable tableName="oauth_refresh_token">
             <column name="token_id" type="varchar(255)"/>

--- a/app/templates/src/main/resources/config/liquibase/changelog/_initial_schema.xml
+++ b/app/templates/src/main/resources/config/liquibase/changelog/_initial_schema.xml
@@ -2,8 +2,7 @@
 <databaseChangeLog
     xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd"
-    objectQuotingStrategy="QUOTE_ALL_OBJECTS">
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
 
     <property name="now" value="now()" dbms="mysql,h2"/>
     <property name="now" value="current_timestamp" dbms="postgresql"/>

--- a/app/templates/src/main/resources/config/mongeez/authorities.xml
+++ b/app/templates/src/main/resources/config/mongeez/authorities.xml
@@ -1,8 +1,8 @@
 <mongoChangeLog>
     <changeSet changeId="ChangeSet-1" author="jhipster">
         <script>
-            db.JHI_AUTHORITY.insert({"_id" : "ROLE_ADMIN"});
-            db.JHI_AUTHORITY.insert({"_id" : "ROLE_USER"});
+            db.jhi_authority.insert({"_id" : "ROLE_ADMIN"});
+            db.jhi_authority.insert({"_id" : "ROLE_USER"});
         </script>
     </changeSet>
 </mongoChangeLog>

--- a/app/templates/src/main/resources/config/mongeez/users.xml
+++ b/app/templates/src/main/resources/config/mongeez/users.xml
@@ -1,9 +1,9 @@
 <mongoChangeLog>
     <changeSet changeId="ChangeSet-2" author="jhipster">
         <script>
-            db.JHI_USER.ensureIndex({"login" : 1});
-            db.JHI_USER.ensureIndex({"email" : 1});
-            db.JHI_USER.insert(
+            db.jhi_user.ensureIndex({"login" : 1});
+            db.jhi_user.ensureIndex({"email" : 1});
+            db.jhi_user.insert(
             {
                 "_id" : "user-0",
                 "login" : "system",
@@ -17,7 +17,7 @@
                 "created_date": new Date(),
                 "authorities": [{"_id": "ROLE_ADMIN"}, {"_id": "ROLE_USER"}]
             });
-            db.JHI_USER.insert(
+            db.jhi_user.insert(
             {
                 "_id" : "user-1",
                 "login" : "anonymousUser",
@@ -31,7 +31,7 @@
                 "created_date": new Date(),
                 "authorities": []
             });
-            db.JHI_USER.insert(
+            db.jhi_user.insert(
             {
                 "_id" : "user-2",
                 "login" : "admin",
@@ -45,7 +45,7 @@
                 "created_date": new Date(),
                 "authorities": [{"_id": "ROLE_ADMIN"}, {"_id": "ROLE_USER"}]
             });
-            db.JHI_USER.insert(
+            db.jhi_user.insert(
             {
                 "_id" : "user-3",
                 "login" : "user",

--- a/entity/index.js
+++ b/entity/index.js
@@ -1349,7 +1349,7 @@ EntityGenerator.prototype.copyEnumI18n = function(language, enumInfo) {
 };
 
 EntityGenerator.prototype.getTableName = function(value) {
-    return _s.underscored(value).toUpperCase();
+    return _s.underscored(value).toLowerCase();
 };
 
 EntityGenerator.prototype.getColumnName = function(value) {

--- a/entity/index.js
+++ b/entity/index.js
@@ -1164,7 +1164,7 @@ EntityGenerator.prototype.files = function files() {
 
     this.entityClass = _s.capitalize(this.name);
     this.entityInstance = _s.decapitalize(this.name);
-    this.entityTableName = _s.underscored(this.name).toUpperCase();
+    this.entityTableName = _s.underscored(this.name).toLowerCase();
 
     this.differentTypes = [this.entityClass];
     if (this.relationships == undefined) {

--- a/entity/templates/src/main/resources/config/liquibase/changelog/_added_entity.xml
+++ b/entity/templates/src/main/resources/config/liquibase/changelog/_added_entity.xml
@@ -68,7 +68,7 @@
                                  referencedColumnNames="id"
                                  referencedTableName="<%
                                  var otherEntityTable = getTableName(relationships[relationshipId].otherEntityName);
-                                 if (otherEntityTable == 'USER') { %>JHI_USER<% } else { %><%=otherEntityTable %><% } %>"/>
+                                 if (otherEntityTable == 'USER') { %>jhi_user<% } else { %><%=otherEntityTable %><% } %>"/>
 <% } else if ((relationships[relationshipId].relationshipType == 'many-to-many') && (relationships[relationshipId].ownerSide == true)) {
     var relationshipName = relationships[relationshipId].relationshipName,
     otherEntityName = relationships[relationshipId].otherEntityName,
@@ -106,7 +106,7 @@
                                  referencedColumnNames="id"
                                  referencedTableName="<%
                                  var otherEntityTable = getTableName(otherEntityName);
-                                 if (otherEntityTable == 'USER') { %>JHI_USER<% } else { %><%=otherEntityTable %><% } %>"/>
+                                 if (otherEntityTable == 'USER') { %>jhi_user<% } else { %><%=otherEntityTable %><% } %>"/>
         <% } %><% } %>
     </changeSet>
 </databaseChangeLog>

--- a/entity/templates/src/main/resources/config/liquibase/changelog/_added_entity.xml
+++ b/entity/templates/src/main/resources/config/liquibase/changelog/_added_entity.xml
@@ -68,7 +68,7 @@
                                  referencedColumnNames="id"
                                  referencedTableName="<%
                                  var otherEntityTable = getTableName(relationships[relationshipId].otherEntityName);
-                                 if (otherEntityTable == 'USER') { %>jhi_user<% } else { %><%=otherEntityTable %><% } %>"/>
+                                 if (otherEntityTable == 'user') { %>jhi_user<% } else { %><%=otherEntityTable %><% } %>"/>
 <% } else if ((relationships[relationshipId].relationshipType == 'many-to-many') && (relationships[relationshipId].ownerSide == true)) {
     var relationshipName = relationships[relationshipId].relationshipName,
     otherEntityName = relationships[relationshipId].otherEntityName,
@@ -106,7 +106,7 @@
                                  referencedColumnNames="id"
                                  referencedTableName="<%
                                  var otherEntityTable = getTableName(otherEntityName);
-                                 if (otherEntityTable == 'USER') { %>jhi_user<% } else { %><%=otherEntityTable %><% } %>"/>
+                                 if (otherEntityTable == 'user') { %>jhi_user<% } else { %><%=otherEntityTable %><% } %>"/>
         <% } %><% } %>
     </changeSet>
 </databaseChangeLog>

